src/Models/Interfaces.ts

```diff
@@ -0,0 +1,19 @@
+import { IBatteryControllerReceivable, IClimateControllerReceivable, IMotorControllerReceivable } from './Controllers';
+import { ControllerTypes } from './Enums';
+
+/**
+ * Message format of received socket messages
+ */
+export interface ISocketMessageReceivable {
+    controller: ControllerTypes;
+    data: IBatteryControllerReceivable | IMotorControllerReceivable | IClimateControllerReceivable;
+}
+
+/**
+ * Message format of transmitted socket messages
+ */
+export interface ISocketMessageTransmittable {
+    controller: ControllerTypes;
+    data: any;
```

Comment:
We should eventually refactor this into a generic type with different messages

Reply:
Fully agreed - I can go in and fix it in a bit. At the moment it works but it definitely needs to be cleaned up.